### PR TITLE
Readd review data

### PR DESCRIPTION
**Pull request by dan-0**
Readds review data, now backed by the ViewModel

Also breaks some of the extensions out into more specific files.

Screen Shot 2020-04-18 at 5 31 28 PM

### CODE REVIEW

File: app/src/main/java/com/rockthevote/grommet/util/extensions/AddressData.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.rockthevote.grommet.util.extensions
+
+import com.rockthevote.grommet.databinding.ViewAddressBinding
+import com.rockthevote.grommet.ui.registration.address.AddressData
+import com.rockthevote.grommet.ui.views.AddressView
+
+fun AddressData.toFriendlyString(): String {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/util/extensions/AddressData.kt:7]: new

File: app/src/main/java/com/rockthevote/grommet/util/extensions/List.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.rockthevote.grommet.util.extensions
+
+fun List<String?>.listOfNotNullOrEmpty(): List<String> = mapNotNull {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/util/extensions/List.kt:3]: new

File: app/src/main/java/com/rockthevote/grommet/util/extensions/RegistrationData.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.rockthevote.grommet.util.extensions
+
+import com.rockthevote.grommet.ui.registration.RegistrationData
+import com.rockthevote.grommet.ui.registration.review.ReviewAndConfirmStateData
+import com.rockthevote.grommet.util.Dates
+
+fun RegistrationData.toReviewAndConfirmStateData(): ReviewAndConfirmStateData {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/util/extensions/RegistrationData.kt:7]: new

File: app/src/main/java/com/rockthevote/grommet/ui/registration/review/ReviewExt.kt [hunk 4]
```diff
@@ -28,4 +29,30 @@ fun FragmentReviewAndConfirmBinding.toReviewData(): ReviewData {
         signature,
         lang.toString()
     )
+}
+
+fun FragmentReviewAndConfirmBinding.bindReviewData(data: ReviewAndConfirmStateData) {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/ui/registration/review/ReviewExt.kt:34]: new

File: app/src/main/java/com/rockthevote/grommet/util/extensions/AddressData.kt [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    return listOf(
+        lineOne,
+        lineTwo,
+        lineThree,
+        lineFour
+    ).listOfNotNullOrEmpty()
+        .joinToString("\n")
+}
+
+fun AddressData.bindToAddressView(view: AddressView) {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/util/extensions/AddressData.kt:32]: moved

File: app/src/main/java/com/rockthevote/grommet/util/extensions/AddressView.kt [hunk 6]
```diff
@@ -0,0 +1,32 @@
+package com.rockthevote.grommet.util.extensions
+
+import com.rockthevote.grommet.databinding.ViewAddressBinding
+import com.rockthevote.grommet.ui.registration.address.AddressData
+import com.rockthevote.grommet.ui.views.AddressView
+
+fun AddressView.toAddress(): AddressData? {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/util/extensions/AddressView.kt:7]: moved

File: app/src/main/java/com/rockthevote/grommet/util/extensions/NameView.kt [hunk 7]
```diff
@@ -0,0 +1,32 @@
+package com.rockthevote.grommet.util.extensions
+
+import com.rockthevote.grommet.data.db.model.Prefix
+import com.rockthevote.grommet.data.db.model.Suffix
+import com.rockthevote.grommet.databinding.ViewNameBinding
+import com.rockthevote.grommet.ui.registration.name.PersonNameData
+import com.rockthevote.grommet.ui.views.NameView
+
+@Throws(IllegalArgumentException::class)
+fun NameView.toPersonName(): PersonNameData? {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/util/extensions/NameView.kt:10]: moved

File: app/src/main/java/com/rockthevote/grommet/util/extensions/PersonNameData.kt [hunk 8]
```diff
@@ -0,0 +1,31 @@
+package com.rockthevote.grommet.util.extensions
+
+import com.rockthevote.grommet.databinding.ViewNameBinding
+import com.rockthevote.grommet.ui.registration.name.PersonNameData
+import com.rockthevote.grommet.ui.views.NameView
+
+fun PersonNameData.bindToNameView(view: NameView) {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/util/extensions/PersonNameData.kt:7]: moved

File: app/src/main/java/com/rockthevote/grommet/util/extensions/String.kt [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package com.rockthevote.grommet.util.extensions
+
+import android.text.Editable
+
+fun String.toEditable(): Editable = Editable.Factory.getInstance().newEditable(this)
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/util/extensions/String.kt:5]: moved

File: app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationViewModel.kt [hunk 10]
```diff
@@ -29,12 +31,22 @@ class RegistrationViewModel(
...
     private val registrationDao: RegistrationDao
 ) : ViewModel() {
+
     private val _registrationData = MutableLiveData(RegistrationData())
     val registrationData: LiveData<RegistrationData> = _registrationData
 
     private val _registrationState: MutableLiveData<RegistrationState> = MutableLiveData(RegistrationState.Init)
     val registrationState: LiveData<RegistrationState> = _registrationState
 
+    private val _reviewAndConfirmState = MediatorLiveData<ReviewAndConfirmState>().apply {
```
Comment (huttneab) [inline app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationViewModel.kt:41]: some new live data type?

**Pull request merged**